### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by ab-noori**
### 🌟Features:

> - [x] - A method to access the current user's data has been implemented in the ApplicationController.
> - [x] - Forms have been created to enable users to compose and submit new posts on their behalf.
> - [x] - The functionality to add comments authored by the current user to posts has been integrated.
> - [x] - Users are now able to express their likes for posts using the provided interface.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -8,4 +8,24 @@ def show
...
+
+  def create
+    @post = @current_user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(user_id: @current_user.id, id: @post.id)
+    else
+      render :new
+    end
+  end
```
Comment (elisha2kyakpo1) [inline app/controllers/posts_controller.rb]: - [ ] I couldn't create a post using the form which is being rendered using the new action. The action is rolled back without any error message. Kindly fix this issue and please for a better user experience, add buttons for adding a post, comment, and like.

rollback

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ab-noori, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Your commit messages a descriptive👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (ab-noori) [reply app/controllers/posts_controller.rb]: **Hi @elisha2kyakpo1 👋,**
Thank you so much 🙏 for reviewing my pull request.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

thank-you-gif-22

**Review by ab-noori — COMMENTED** [review]
**Hi @elisha2kyakpo1 👋,**
Thank you so much 🙏 for reviewing my pull request.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

thank-you-gif-22

**Review by rotimiazeez — APPROVED** [review]
Hi @ab-noori 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* ApplicationController contains `#current_user` method  ✔️
* The posts controller has new and create actions ✔️
* The comments controller has new and create actions ✔️
* The likes controller has a create actions ✔️
* There is a route in the config/routes.rb file for creating posts, likes, comments. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @rotimiazeez 👋,**
Thank you so much 🙏  for reviewing and approving my pull request.
Your nice feedback is really appreciated 🙏.

Thank-You-Very-Much

**Pull request merged**